src/l500/l500-options.cpp

```diff
@@ -255,69 +368,147 @@ namespace librealsense
         return res;
     }
 
-    void l500_options::on_set_option(rs2_option opt, float value)
+    rs2_l500_visual_preset l500_options::calc_preset_from_controls()
     {
-        if (opt == RS2_OPTION_VISUAL_PRESET)
+        try
         {
-            verify_max_usable_range_restrictions( opt, value );
-            change_preset(static_cast<rs2_l500_visual_preset>(int(value)));
-        }
-        else
-        {
-            auto advanced_controls = get_advanced_controls();
-            if (std::find(advanced_controls.begin(), advanced_controls.end(), opt) != advanced_controls.end())
+            // compare default values to current values 
+            // exept from laser power that can get diffrant value according to preset
+            std::map< rs2_option, float > hw_options_default_values;
+            for( auto control : _hw_options )
             {
-                verify_max_usable_range_restrictions( opt, value );
-                move_to_custom();
+                if( control.first != RS2_OPTION_LASER_POWER )
+                {
+                    if( control.second->get_range().def != control.second->query() )
+                        return RS2_L500_VISUAL_PRESET_CUSTOM;
+                }
             }
-            else
-                throw  wrong_api_call_sequence_exception(to_string() << "on_set_option support advanced controls only "<< opt<<" injected");
+
+                // all the hw_options values are equal to their default values
+                // now what is left to check if the gain and laser power correspond to one of the
+                // presets
+                auto max_laser = _hw_options[RS2_OPTION_LASER_POWER]->get_range().max;
+                auto def_laser = _hw_options[RS2_OPTION_LASER_POWER]->get_range().def;
+
+                std::map< std::pair< rs2_digital_gain, float >, rs2_l500_visual_preset >
+                    gain_and_laser_to_preset = {
+                        { { RS2_DIGITAL_GAIN_HIGH, def_laser }, RS2_L500_VISUAL_PRESET_NO_AMBIENT },
+                        { { RS2_DIGITAL_GAIN_LOW, max_laser }, RS2_L500_VISUAL_PRESET_LOW_AMBIENT },
```

Comment:
`LOW_AMBIENT` and `SHORT_RANGE` are the same, we think:
Need a test that this is true
Need to check that this is true both with FW < 1.5.3 and FW 1.5.3+
Need to remove from here the one that we don't want to show (is short-range the one we want to show?)